### ISSUE TITLE
RJSF vertical layout

### ISSUE THREAD

**Issue by fabcor-maxiv** [mxcube/mxcube3#1234]
Anyone knows how to get a react-jsonschema-form (RJSF) form in a vertical layout?

I have seen in some places `"ui:widget": "vertical_box"`, but it does not seem to have any effect (not that I could see, and I might be doing something wrong):

https://github.com/mxcube/mxcubecore/blob/644f6eaa15da57e7ffbb5b74fe421af90c497670/mxcubecore/HardwareObjects/Gphl/GphlWorkflow.py#L532

**Comment by fabcor-maxiv** [comment]
This is what we get:

![image](https://github.com/mxcube/mxcubeweb/assets/140523662/7e183092-aa43-41bd-8e4d-2cfd9accd5f7)

Which is surprising since the default behavior of RJSF seems to be to do vertical layout.

We are not happy with this horizontal layout as seen in the screenshot above. I guess, we would be happy with the default RJSF vertical layout. Ideally we would like a grid layout as can be seen in the screenshot at the following link: <https://mxcubecore.readthedocs.io/en/stable/dev/json-schema-generated-user-interface.html#precharacterisation>.

![image](https://github.com/mxcube/mxcubeweb/assets/140523662/c6c2d73b-938d-4478-97d7-92a07b5149e1)

Any advice? Is it possible at all in MXCuBE-Web or is it a feature in MXCuBE-Qt only?

@rhfogh @axelboc @elmjag

**Comment by marcus-oscarsson** [comment]
As we discussed last time the feature for generating these dialogs are still quite basic and still under development in mxcubeweb. The current solution should render the inputs in vertically in two columns, so that it looks more or less like the other dialogs. I'm a bit surprised by your screen shot. I have to double check but we perhaps made some additional changes that we have not pushed yet.

There are still a few things to continue to develop related to this and its:

   - Dropping RJSF 
   - Rendering of categories (or several models)
   - Handling of commonly used input like space group

**Comment by fabcor-maxiv** [comment]
For the record, this is how "TestCollection" looks like for me right now:

![image](https://github.com/mxcube/mxcubeweb/assets/140523662/0c95432d-c07c-4ddc-954d-002c1fb2f0c1)

**Comment by marcus-oscarsson** [comment]
That does indeed look strange, it should be two columns.

**Comment by marcus-oscarsson** [comment]
I pushed a change we did a while back ago that we have still not pushed. There where some changes in RJSF a while back and we needed to adapt to those. Please have a look, I think it solves your issue.

  https://github.com/mxcube/mxcubeweb/tree/mo-task-dialog

**Comment by fabcor-maxiv** [comment]
> I pushed a change we did a while back ago that we have still not pushed. There where some changes in RJSF a while back and we needed to adapt to those. Please have a look, I think it solves your issue.
> 
> https://github.com/mxcube/mxcubeweb/tree/mo-task-dialog

Yes, I ended up with nearly the same patch/diff yesterday. I was not sure about the `asyncValidate` part, though.

Thanks, we will try that : )

**Comment by elmjag** [comment]
I have tested cherry-picking this commit: https://github.com/mxcube/mxcubeweb/commit/2a56fc882767ad888fc68b40f8a46810fbeab075 It does the trick for us. The dialogs are now rendered with a 'two column' layout. Thanks!

I guess merging that commit is a solution to this issue.

**Comment by marcus-oscarsson** [comment]
:+1: Thanks for letting me know, Ill remove the draft then
